File: src/06.hashMap/16.219.ts
export function containsNearbyDuplicate(nums: number[], k: number): boolean {
    const map = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const item = nums[i];
        if (map.has(item)) {
            if (Math.abs(map.get(item) - i) <= k) {
                return true;
            }
        }

        map.set(item, i);
    }

    return false;
}

File: src/06.hashMap/16.219.go
package hashmap

import "math"

func ContainsNearbyDuplicate(nums []int, k int) bool {
	numsMap := map[int]int{}

	for i := 0; i < len(nums); i++ {
		if v, ok := numsMap[nums[i]]; ok {
			if int(math.Abs(float64(v-i))) <= k {
				return true
			}
		}

		numsMap[nums[i]] = i
	}

	return false
}
